src/app/components/primary-navbar/primary-navbar.component.html

```diff
@@ -1 +1,159 @@
-<p>primary-navbar works!</p>
+<ng-template #desktop>
+    <div class="desktop-navbar">
+        <div class="nav-wrapper">
+            <div class="logo">
+                <span class="logo-image"></span>
+                <span id="clark-logo" class="logo-link" [routerLink]="['home']">CLARK</span>
+            </div>
+            <div class="mid-bar">
+                <div class="level-picker" #academicLevelsAnchor (click)="dropdowns.toggleLevelsDropdown()">
+                    All Academic Levels
+                    <span class="arrow" *ngIf="!levelsDropdown">
+                        <i class="fas fa-chevron-down"></i>
+                    </span>
+                    <span class="arrow" *ngIf="levelsDropdown">
+                        <i class="fas fa-chevron-up"></i>
+                    </span>
+                </div>
+                <clark-context-menu *ngIf="levelsDropdown" [anchor]="academicLevelsAnchor" (close)="dropdowns.closeAll()">
+                    <div #contextMenu class="context-popup">
+                        <ul>
+                            <li *ngFor="let l of academicLevels" [routerLink]="[]">{{ l }}</li>
+                        </ul>
+                    </div>
+                </clark-context-menu>
+                <div class="search-bar">
+                    <clark-search></clark-search>
+                </div>
+            </div>
+            <div class="right">
+                <button class="contribute-button" *ngIf="isLoggedIn">
```

Comment:
sure, sorry about that, there were a few things like that, where it wasn't clear what the button was supposed to do, so in the moment I moved on and decided to come back to it but I forgot. a little explanation on shortcut about what everything is actually supposed to do when you click it would be great